salt/states/archive.py

```diff
@@ -1174,6 +1175,12 @@ def extracted(name,
     created_destdir = False
 
     if extraction_needed:
+        if source_is_local and source_hash and not skip_verify:
+            ret['result'] = __salt__['file.check_hash'](source_match, source_sum['hsum'])
+            if not ret['result']:
+                ret['comment'] = 'Hash sum mismatch for {0}!'.format(source_match)
```

Comment:
The exclamation point is unnecessary here. Also, "hash sum mismatch" might not mean anything to the person running the state, since we don't use that terminology anywhere. Can we change this to something like the following?

```python
                ret['comment'] = \
                    '{0} does not match the desired source_hash {1}'.format(
                        source_match, source_sum['hsum']
                    )
```